src/store/workbenchReducer.ts

```diff
@@ -1,14 +1,26 @@
 import { PayloadAction } from '@reduxjs/toolkit';
 import { IColumnDesc } from 'lineupjs';
 import { IRow } from 'tdp_core';
-import { EWorkbenchDirection, ISelectedMapping, IWorkbench } from './interfaces';
+import { EWorkbenchDirection, ISelectedMapping, IWorkbench, IOrdinoAppState } from './interfaces';
 
 export const workbenchReducers = {
-  addFirstWorkbench(state, action: PayloadAction<IWorkbench>) {
+  addFirstWorkbench(
+    state,
+    action: PayloadAction<{
+      workbench: IWorkbench;
+      globalQuery: IOrdinoAppState['globalQuery'];
+      appliedQueryFilter: IOrdinoAppState['appliedQueryFilter'];
+    }>,
+  ) {
     state.focusWorkbenchIndex = 0;
-    state.workbenches = [action.payload];
+    state.workbenches = [action.payload.workbench];
+    state.globalQuery = action.payload.globalQuery;
+    state.appliedQueryFilter = action.payload.appliedQueryFilter;
   },
   addWorkbench(state, action: PayloadAction<IWorkbench>) {
+    // always add the global query to the next workbench
+    // by default the first view will be a ranking view that can deal with the global query parameter
+    Object.assign(action.payload.views[0].parameters, { globalQuery: state.globalQuery, appliedQueryFilter: state.appliedQueryFilter });
```

Comment:
The comment is really helpful, thanks for that.

Patching the first view parameters looks dangerous to me because we assume (as you wrote) that the workbench has a ranking view (which might not be the case at one point). 

Hence, I would suggest to add this directly to the first view definition when calling `addWorkbench` in CreateNextWorkbenchSidebar.tsx:

https://github.com/Caleydo/ordino/blob/ba6c6422c5e139b96380a014f8dcb8b41a9eb947/src/app/workbench/sidebar/CreateNextWorkbenchSidebar.tsx#L100-L108

Same would go for `addFirstWorkbench` above in EntitySelectionCard.tsx:

https://github.com/datavisyn/reprovisyn/blob/581352a3fcf9757ba0883240383ee5a294639f31/src/views/EntitySelectionCard.tsx#L22

What do you think about this?

Reply:
Beside the comment above: Do we even use the `globalQuery` from the first view parameter? Because the [GenericRanking.tsx](https://github.com/datavisyn/reprovisyn/blob/5c5959a3c994d748d75c7117e6525b84a338256c/src/views/GenericRanking.tsx#L14) uses only the `appliedQueryFilter`.

Reply:
Your first suggestion was great, I changed the location where the parameters are added.